python/cudf/cudf/core/column/column.py

```diff
@@ -990,6 +990,13 @@ def astype(self, dtype: Dtype, **kwargs) -> ColumnBase:
             np.object_,
             str,
         }:
+            if cudf.get_option("mode.pandas_compatible") and np.dtype(
+                dtype
+            ).type in {np.object_}:
+                raise ValueError(
+                    f"{dtype} is an unsupported dtype to type-cast to, use "
```

Comment:
This simplifies the error text. Might require re-wrapping.
```suggestion
                    f"Casting to {dtype} is not supported, use "
```